src/GovUk.Education.ExploreEducationStatistics.Admin.Tests/Services/Methodologies/MethodologyImageServiceTests.cs

```diff
@@ -129,7 +130,101 @@ await using (var contentDbContext = InMemoryApplicationDbContext(contentDbContex
+                    Type = Image
+                }
+            };
+
+            var imageFile2 = new MethodologyFile
+            {
+                Methodology = methodology,
+                File = new File
+                {
+                    RootPath = Guid.NewGuid(),
+                    Filename = "image2.png",
+                    Type = Image
+                }
+            };
+            
+            var imageFile2UsedByAnotherMethodology = new MethodologyFile
+            {
+                Methodology = anotherMethodology,
+                File = imageFile2.File
+            };
+
+            var contentDbContextId = Guid.NewGuid().ToString();
+
+            await using (var contentDbContext = InMemoryApplicationDbContext(contentDbContextId))
+            {
+                await contentDbContext.Methodologies.AddAsync(methodology);
+                await contentDbContext.MethodologyFiles.AddRangeAsync(imageFile1, imageFile2,
+                    imageFile2UsedByAnotherMethodology);
+                await contentDbContext.SaveChangesAsync();
+            }
+
+            var blobStorageService = new Mock<IBlobStorageService>(MockBehavior.Strict);
+
+            blobStorageService.Setup(mock =>
+                    mock.DeleteBlob(PrivateMethodologyFiles, imageFile1.Path()))
+                .Returns(Task.CompletedTask);
+
+            await using (var contentDbContext = InMemoryApplicationDbContext(contentDbContextId))
+            {
+                var service = SetupMethodologyImageService(contentDbContext: contentDbContext,
+                    blobStorageService: blobStorageService.Object);
+
+                var result = await service.UnlinkAndDeleteIfOrphaned(methodology.Id, new List<Guid>
+                {
+                    imageFile1.File.Id,
+                    imageFile2.File.Id
+                });
+
+                Assert.True(result.IsRight);
+
+                blobStorageService.Verify(mock =>
+                    mock.DeleteBlob(PrivateMethodologyFiles, imageFile1.Path()), Times.Once);
+            }
+
+            await using (var contentDbContext = InMemoryApplicationDbContext(contentDbContextId))
+            {
+                // All of the MethodologyFiles links for the methodology having the images removed from it are removed.
+                Assert.Null(await contentDbContext.MethodologyFiles.FindAsync(imageFile1.Id));
+                Assert.Null(await contentDbContext.Files.FindAsync(imageFile1.File.Id));
+                Assert.Null(await contentDbContext.MethodologyFiles.FindAsync(imageFile2.Id));
+                
+                // However, the file that is still lined to anotherMethodology remains in the File table, as does
+                // its link to anotherMethodology.
+                Assert.NotNull(await contentDbContext.MethodologyFiles.FindAsync(imageFile2UsedByAnotherMethodology.Id));
+                
+                var fileIdForFileLinkedToDeletingMethodology = 
```

Comment:
I think these should be files not fileId in their names

Reply:
Good spot!